HOK.Core/HOK.Core/BackgroundTasks/Rules.cs

```diff
@@ -0,0 +1,161 @@
+                        hokPrintSetParam.SetValue(new IntegerParameterValue(1));
+                        tr.Commit();
+                    }
+                }
+            }
+            hokPrintSetParamId = GlobalParametersManager.FindByName(
+                doc,
+                "Add ALL Sheets to HOK Print Set"
+            );
+            var hokPrintSetParameter = doc.GetElement(hokPrintSetParamId) as GlobalParameter;
+            if ((hokPrintSetParameter.GetValue() as IntegerParameterValue).Value == 1)
+            {
+                goto AddAllSheets;
+            }
+            else
+            {
+                return;
+            }
+
+        AddAllSheets:
+            PrintManager pm = doc.PrintManager;
+
+            // Ensure that the print manager setting is set to selected views
+            if (pm.PrintRange != PrintRange.Select)
+            {
+                pm.PrintRange = PrintRange.Select;
+            }
+
+            ViewSheetSetting existingVSS = pm.ViewSheetSetting;
+            // Save the previously selected viewsheet set so we can set it back to the original at the end
+            var existingViewSheetSet = existingVSS.CurrentViewSheetSet;
+
+            var hokSheetSet = new FilteredElementCollector(doc)
+                .OfClass(typeof(ViewSheetSet))
+                .Cast<ViewSheetSet>();
+
+            if (hokSheetSet.Select(ss => ss.Name).Contains("_HOK Automated Print Set"))
+                existingVSS.CurrentViewSheetSet = hokSheetSet.First(
+                    ss => ss.Name == "_HOK Automated Print Set"
+                );
+
+            List<ElementId> sheetsInModel = new FilteredElementCollector(doc)
+                .OfClass(typeof(ViewSheet))
+                .ToElements()
+                .Cast<ViewSheet>()
+                .Select(s => s.Id)
+                .ToList();
+
+            // Add the excluded sheets to the sheet set
+            ViewSet newVSS = new ViewSet();
+            foreach (var viewId in sheetsInModel)
+            {
+                View currView = (View)doc.GetElement(viewId);
+                if (!newVSS.Contains(currView))
+                {
+                    newVSS.Insert(currView);
+                }
+                else
+                {
+                    continue;
+                }
+            }
+            using (Transaction tr = new Transaction(doc, "Add Sheets to View Sheet Set"))
+            {
+                if (doc.IsModifiable)
+                {
+                    if (
+                        hokSheetSet != null
+                        && ((ViewSheetSet)existingVSS.CurrentViewSheetSet).Name
+                            == "_HOK Automated Print Set"
+                    )
+                    {
+                        try
+                        {
+                            existingVSS.CurrentViewSheetSet.Views = newVSS;
+                            existingVSS.Save();
+                        }
+                        catch { }
```

Comment:
Since we're wrapping the code where this method runs in a try/catch, feel free to add an Exception with a message so it'll get logged